### ISSUE TITLE
Deserialising unicode escape gives non-UTF8 String

### ISSUE THREAD

**Issue by 5225225** [simd-lite/simd-json#228]
```rust
fn main() {
    let mut data = *br#""\uDE71""#; // [34, 92, 117, 68, 69, 55, 49, 34];

    let x: String = simd_json::from_slice(&mut data).unwrap();
    String::from_utf8(x.into_bytes()).unwrap();
}
```

Here, the second unwrap here fails, which means the library gave me a String with invalid UTF-8.

**Comment by Licenser** [comment]
@hkratz I think this tracks back to simdutf8, 

I added the following test case
```
    test_invalid_after_prefixes(br#""\uDE71""#, 0, None, Some(5));
```
and it doesn't throw an error.

**Comment by hkratz** [comment]
Simdutf8 usage in simd-json only confirms that the input byte sequence _without any escape processing_ is valid UTF-8. Here the input byte sequence `[34, 92, 117, 68, 69, 55, 49, 34]` is valid UTF-8. AFAICS unicode escape processing is done by simd-json itself here: https://github.com/simd-lite/simd-json/blob/6af8b2a1cc3f9f10fcc11624bcd1354d506a6807/src/stringparse.rs#L22-L73

This does not seem to handle a lone low surrogates such as `\uDE71` at all (which should result in an error).

**Comment by Licenser** [comment]
Heya, sorry about that, too many distractions these days, you're completely right, I missed that there is a \u in there that is not escaped during the test itself 🤦

**Comment by Licenser** [comment]
Heya @5225225 sorry this took so long, I was a bit too all over the place and totally missed the real issue here ... on the bright side, it should be fixed now and released as 0.5.1 :)
